syntax = "proto3";

// 股票基本信息获取和更新
package api.crawler.v1;
import "google/api/annotations.proto";
import "api/crawler/v1/index.proto";

option go_package = "gostock/api/crawler/v1;v1";
option java_multiple_files = true;
option java_package = "api.crawler.v1";


service StockService {
	rpc GetStockInfo (StockBasicRequest) returns (StockBasic) {
		option (google.api.http) = {
			get: "/stock/{id}"
		};
	};
    rpc GetIndexInfo (IndexRequest) returns (Index) {
		option (google.api.http) = {
			get: "/index/{id}"
		};
	};
}


message StockBasicRequest {
    int32  id = 1;
    string is_hs = 2;  // str	N	是否沪深港通标的，N否 H沪股通 S深股通
    string list_status = 3; //	N	上市状态 L上市 D退市 P暂停上市，默认是L
    string exchange	= 4; //	N	交易所 SSE上交所 SZSE深交所 BSE北交所
    string ts_code	= 5; //	N	TS股票代码
    string market	= 6; //	N	市场类别 （主板/创业板/科创板/CDR/北交所）
    int32 limit	= 7;     //	N	
    int32 offset	= 9; //	N	
    string name	= 10;     //	N	名称
}

message StockBasic {
    int32  id = 1;
    string ts_code = 2; // Y	TS代码
    string symbol	 = 3; //	Y	股票代码
    string name	 = 4; //	Y	股票名称
    string area	 = 5; //	Y	地域
    string industry	 = 6; //	Y	所属行业
    string fullname	 = 7; //	N	股票全称
    string enname = 8; //	N	英文全称
    string cnspell = 9; //	N	拼音缩写
    string market = 10; //	Y	市场类型（主板/创业板/科创板/CDR）
    string exchange	 = 11; //	N	交易所代码
    string curr_type = 12; //	N	交易货币
    string list_status = 13; //	N	上市状态 L上市 D退市 P暂停上市
    string list_date = 14; //	Y	上市日期
    string delist_date = 15; //	N	退市日期
    string is_hs = 16; //	N	是否沪深港通标的，N否 H沪股通 S深股通
}